libs/error-stack/src/frame/mod.rs

```diff
@@ -90,6 +92,72 @@ impl Frame {
         self.frame.as_any_mut().downcast_mut()
     }
 
+    /// `replace_with_printable`: if true, `Debug` and `Display` implementations will be eagerly
+    ///     evaluated and stored in place of the taken type,
+    ///     can be set to false when the `Report` will no longer be used to avoid the cost of
+    ///     formatting
+    #[must_use]
+    pub(crate) fn downcast_take<T: Send + Sync + 'static>(
```

Comment:
As mentioned above, we should not add this as part of this PR. This and any other related work should be removed from this PR to make this PR focused on `Report<dyn Error>`.

Also, the behavior to replace the context with another opaque context feels like an anti pattern. If we want such a functionality, the new context should be part of the signature, and this could be defined on any `Report` (e.g. `report.replace_context(MyContext)`. Let's remove it for now.